File: LeetCodeCPlusPlus/py73.cpp
// 矩阵置0
#include <vector>
#include <string>
#include <unordered_set>
using namespace std;

class Solution {
public:
    void setZeroes(vector<vector<int>>& matrix) {
        unordered_set<int> row, col;
        int n = matrix.size();
        if (n == 0)
            return;
        int m = matrix[0].size();
        for (int i=0;i<n;i++)
            for (int j=0;j<m;j++)
                if (matrix[i][j] == 0) {
                    row.insert(i);
                    col.insert(j);
                }
        for (int i=0;i<n;i++)
            for (int j=0;j<m;j++)
                if (row.count(i) == 1 || col.count(j) == 1) 
                    matrix[i][j] = 0;
        
    }
};

File: LeetCodePython/py73.py
# 矩阵置0
from typing import List
class Solution:
    def setZeroes(self, matrix: List[List[int]]) -> None:
        """
        Do not return anything, modify matrix in-place instead.
        """
        n = len(matrix)
        if n==0:
            return
        m = len(matrix[0])
        row = set()
        col = set()
        for i in range(n):
            for j in range(m):
                if matrix[i][j] == 0:
                    row.add(i)
                    col.add(j)
        
        for i in range(n):
            for j in range(m):
                if i in row or j in col:
                    matrix[i][j] = 0
        